### PR TITLE
git actions php8

### PR DESCRIPTION
**Pull request by kongtiaowang**
## Brief summary of changes

- [ ] Have you updated related documentation?

#### Testing instructions (if applicable)

1.

#### Link(s) to related issue(s)

* Resolves #  (Reference the issue this fixes, if any.)

### CODE REVIEW

File: .github/workflows/loristest.yml [hunk 1]
```diff
@@ -37,7 +38,7 @@ jobs:
           ${{ runner.os }}-php-
 
     - name: Change PHP Version in Dockerfile
-      run: sed -i "s/7.3/${{ matrix.php }}/g" Dockerfile.test.php7
+      run: sed -i "s/7.4/${{ matrix.php }}/g" Dockerfile.test.php7
```
Comment (driusan) [inline .github/workflows/loristest.yml:41]: Why is this change needed? If the script is doing what it should then it shouldn't matter if it's "7.3", "7.4" or "apple".. this makes me worried that it's not properly testing 8.0.

File: php/libraries/User.class.inc [hunk 2]
```diff
@@ -338,7 +338,7 @@ class User extends UserPermissions implements \LORIS\StudyEntities\AccessibleRes
     {
         $site_list = $this->userInfo['examiner'];
         unset($site_list['pending']);
-        natcasesort($site_list);
+        ksort($site_list);
```
Comment (driusan) [inline php/libraries/User.class.inc:341]: Why is this needed? It's a different sorting order..

File: src/Middleware/UserPageDecorationMiddleware.php [hunk 3]
```diff
@@ -1,4 +1,16 @@
 <?php
+/**
+ * This file handles the UserPageDecorationMiddleware for LORIS
+ *
+ * PHP Version 7
+ *
+ * @category LORIS
+ * @package  Main
+ * @author   Dave MacFarlane <driusan@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
```
Comment (driusan) [inline src/Middleware/UserPageDecorationMiddleware.php:10]: This doesn't have anything to do with PHP8 and looks like a step backwards in terms of our phpcs standards..

**Review by driusan — CHANGES_REQUESTED** [review]

File: php/libraries/User.class.inc (hunk 2 above)
Comment (kongtiaowang) [reply php/libraries/User.class.inc:341]: Site_list is a multiple dimension array. The natcasesort function only works with multiple dimension array in php7.4, not php8.  The ksort will do the same thing with natcasesort. only sorts the first level key.

File: .github/workflows/loristest.yml (hunk 1 above)
Comment (kongtiaowang) [reply .github/workflows/loristest.yml:41]: 7.4 is the default version of the Docker PHP file.
if will be replaced by matrix.php (7.4 or 8.0) .
Screen Shot 2021-02-05 at 10 58 09 AM
Screen Shot 2021-02-05 at 10 57 44 AM

File: php/libraries/User.class.inc (hunk 2 above)
Comment (driusan) [reply php/libraries/User.class.inc:341]: The sorting order is different. ksort sorts by key, while natcasesort sorts more human-like way for the front end (ie. the sorting goes "1", "2", .. "10" rather than "1, "10", "2" ..). But if that's not working with php8 and that's a trade-off we need to make, I guess having PHP8 tested is more important.
